src/main.rs

```diff
@@ -26,16 +27,50 @@ async fn main() -> Result<()> {
 async fn run() -> Result<()> {
     let config_dir = dirs::config_dir().ok_or_else(|| anyhow!("config directory not found"))?;
     let config_file = config_dir.join("bob").join("config.json");
-    let config: Config = match tokio::fs::read_to_string(config_file).await {
-        Ok(config_file) => serde_json::from_str(&config_file)?,
+    let config: Config = handle_config(tokio::fs::read_to_string(config_file).await)?;
+    if let Err(error) = modules::cli::start(config).await {
+        return Err(anyhow!(error));
+    }
+    Ok(())
+}
+
+fn handle_config(config_file: Result<String, std::io::Error>) -> Result<Config> {
+    let config = match config_file {
+        Ok(config_file) => {
+            let mut config: Config = serde_json::from_str(&config_file)?;
+            handle_envars(&mut config)?;
+            config
+        }
         Err(_) => Config {
             enable_nightly_info: None,
             downloads_dir: None,
             installation_location: None,
         },
     };
-    if let Err(error) = modules::cli::start(config).await {
-        return Err(anyhow!(error));
+
+    Ok(config)
+}
+
+fn handle_envars(config: &mut Config) -> Result<()> {
+    let re = Regex::new(r"\$([A-Z_]+)").unwrap();
+
+    if let Some(value) = &config.downloads_dir {
+        let new_value = handle_envar(value, &re)?;
+        config.downloads_dir = Some(new_value);
     }
+
+    if let Some(value) = &config.installation_location {
+        let new_value = handle_envar(value, &re)?;
+        config.installation_location = Some(new_value);
+    }
+
     Ok(())
 }
+
+fn handle_envar(value: &str, re: &Regex) -> Result<String> {
+    let extract = re.captures(value).unwrap().get(1).unwrap().as_str();
```

Comment:
this breaks regular config usage, if the value doesn't conform to the regex, we panic here

Reply:
fixed, thank you for noticing